cli/installer/docker_compose.go

```diff
@@ -166,7 +166,7 @@ func getDockerComposeFileContents(ui UI, config configuration) []byte {
 		ui.Exit(fmt.Sprintf("cannot configure tracetest container: %s", err.Error()))
 	}
 
-	if config.Bool("tracetest.collector.install") {
+	if config.Bool("tracetest.collector.install") && config.Bool("tracetest.backend.install") {
```

Comment:
what issue is this trying to fix? Looks like it will ask the user if they want to install the collector (asked after the backend install), but then it won't based on a previous question answer.

Can you explain the goal of this change?

Reply:
`fixOtelCollectorContainer` relies on `tracetest.backend.endpoint.collector` to inject the JAEGER endpoint to the collector env object. 

I added a change to check for jaeger instead of `tracetest.backend.install`. But the original problem is that `tracetest.backend.endpoint.collector` doesn't exist for all backend storages. In the case of OpenSearch, because this configuration was not available, the CLI would panic due to [this](https://github.com/kubeshop/tracetest/blob/c4d1c1f0763e179eba45ef16ffed80ae3df0a9c9/cli/installer/installer.go#L101)

Reply:
As fixOtelCollectorContainer is only necessary for jaeger installs, checking for the backend type is enough to prevent this issue from happening